File: src/main/java/by/architecture/map/service/ConstructionService.java
package by.architecture.map.service;

import by.architecture.map.dto.ConstructionDto;
import by.architecture.map.entity.Construction;

import java.util.List;

public interface ConstructionService {

    List<Construction> findAll();

    void add(ConstructionDto construction);

    void update(Integer idOfOldConstruction, ConstructionDto updatedConstruction);

    void delete(Integer id);

}

File: src/main/kotlin/by/architecturemap/belarus/service/ConstructionService.kt
package by.architecturemap.belarus.service

import by.architecturemap.belarus.dto.ConstructionDTO
import by.architecturemap.belarus.entity.Construction

interface ConstructionService {

    fun create(construction: Construction): Construction
    fun findAll(): List<Construction>
    fun find(id: Int): Construction

    /**
     * Using for finding Constructions in Elasticsearch
     */
    fun find(constructionDTO: ConstructionDTO): List<Construction>

    fun update(id: Int, updatedConstruction: Construction): Construction
    fun patchUpdate(id: Int, updatedConstruction: Construction): Construction
    fun delete(id: Int)
}
